python_modules/dagster/dagster/_core/definitions/graph_definition.py

```diff
@@ -976,6 +980,97 @@ def get_op_input_output_handle_pairs(
 
         return result
 
+    def with_inner_inputs_mapped_to_outer(
+        self, input_handles: AbstractSet[NodeInputHandle], prefix: Optional[NodeHandle]
+    ) -> Tuple["GraphDefinition", Mapping[NodeInputHandle, str]]:
+        """Create a new graph with the inner inputs mapped to new outer inputs.
+
+        Args:
+            input_handles: Input handles within the graph that are not mapped to inputs within their
+                immediate parent graph.
+
+        Returns:
+            A tuple of the new graph with the inner inputs mapped to new outer inputs, and a mapping
+            from inner input handles to the outer input names.
+        """
+        all_input_names = set(self.input_dict.keys())
+        new_input_mappings: List[InputMapping] = []
+        outer_inputs_by_inner_input_handle: Dict[NodeInputHandle, str] = {}
+        node_defs_by_name: Mapping[str, NodeDefinition] = {}
+        dependencies: Dict[NodeInvocation, Mapping[str, IDependencyDefinition]] = {}
+
+        invocations_by_name = {
+            invocation.alias or invocation.name: invocation
+            for invocation in self.dependencies.keys()
+        }
+
+        for node in self.nodes:
+            for node_input in node.inputs():
+                input_handle = node_input.to_handle(prefix)
+                if input_handle in input_handles:
+                    outer_input_name = make_alias_if_collision(
```

Comment:
this bit sketches me out somewhat because we're iterating over `self.nodes` and `node.inputs()`, both of which come from unordered collections (set and dictionary, respectively).

this makes it seem to me that if we do this subset operation in a subtask, it's possible that it will end up assigning something a different name than the main task did, which would (hopefully) cause an error.

i think to actually test this you'd need to do something with the multi-process executor (such that we're forced call this code multiple times within a single run)

or you could just do some manual runs and see what happens

simple fix would of course just be to sort those iterables before looping